src/components/cylc/gscan/GScan.vue

```diff
@@ -279,14 +267,17 @@ export default {
      * then we apply the filters to the list of workflows.
      */
     searchWorkflows: {
+      deep: true,
       immediate: false,
       handler: 'filterWorkflows'
     },
     workflows: {
+      deep: true,
       immediate: true,
       handler: 'filterWorkflows'
     },
     filteredWorkflows: {
+      deep: true,
```

Comment:
Dammit, does it break without this? I think it's a nasty performance hit.

Reply:
https://v3-migration.vuejs.org/breaking-changes/watch.html

I don't think the deep watch will have any effect for `filteredWorkflows` because it is only ever replaced, not mutated. I'll try removing `deep`

Reply:
Deep means 100% deep as far as I know. However, for `workflows` I think it is pretty much necessary  as part of the functionality is filtering by task status

Reply:
I think the best option is to use intermediate computed properties that pull out the relevant fields and watch those instead. Will take some refactoring

Reply:
Hmm, on second thought the `stateTotals` thinggy includes counts:

```
        "stateTotals": {
          "waiting": 0,
          "expired": 0,
          "preparing": 0,
          "submit-failed": 0,
          "submitted": 0,
          "running": 0,
          "failed": 0,
          "succeeded": 0
        }
```

So I guess task state changes would trigger this anyway because the count would change. But there are other things besides task state changes, job messages, graph edges, etc, which could trigger re-filtering with `deep: true`.

Reply:
This needs investigation/solving. Please put up an issue if not doing it here.

Reply:
#1296